src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/EditAppealAfterSubmitHandler.java

```diff
@@ -75,35 +75,43 @@ public PreSubmitCallbackResponse<AsylumCase> handle(
 
         Optional<OutOfCountryDecisionType> maybeOutOfCountryDecisionType = asylumCase.read(OUT_OF_COUNTRY_DECISION_TYPE, OutOfCountryDecisionType.class);
 
-        if (maybeOutOfCountryDecisionType.isPresent()) {
-            OutOfCountryDecisionType decisionType = maybeOutOfCountryDecisionType.get();
-
-            if (decisionType == OutOfCountryDecisionType.REMOVAL_OF_CLIENT) {
-                Optional<String> maybeHomeOfficeDecisionLetterDate = asylumCase.read(DECISION_LETTER_RECEIVED_DATE);
-
-                decisionDate =
-                    parse(maybeHomeOfficeDecisionLetterDate
-                        .orElseThrow(() -> new RequiredFieldMissingException("decisionLetterReceivedDate is not present")));
-            } else if (decisionType == OutOfCountryDecisionType.REFUSAL_OF_PROTECTION) {
-                Optional<String> maybeDateClientLeaveUk = asylumCase.read(DATE_CLIENT_LEAVE_UK);
-
-                decisionDate =
-                    parse(maybeDateClientLeaveUk
-                        .orElseThrow(() -> new RequiredFieldMissingException("dateClientLeaveUk is not present")));
-            } else if (decisionType == OutOfCountryDecisionType.REFUSAL_OF_HUMAN_RIGHTS) {
-                Optional<String> maybeDateEntryClearanceDecision = asylumCase.read(DATE_ENTRY_CLEARANCE_DECISION);
-
+        if (asylumCase.read(IS_ACCELERATED_DETAINED_APPEAL, YesOrNo.class)
```

Comment:
Did we need to make some changes to this edit appeal event for RIA-6148?

Reply:
@Ryan-Nesbitt can I work through this logic with yourself this morning?